msos/MixedStack.cs

```diff
@@ -312,5 +405,251 @@ public void PrintStackTrace(CommandExecutionContext context, uint index)
+            KernelObjectTypeName = handle.TypeName;
+            //TODO: Convertion
+            Reason = ConvertToUnified(handle.Type);
+            Type = UnifiedBlockingType.DumpHandle;
+            Handle = handle.Handle;
+        }
+
+        internal UnifiedBlockingObject(CRITICAL_SECTION section, ulong handle)
+            : this(OriginSource.StackWalker)
+        {
+            Owners = new List<UnifiedThread>();
+            Owners.Add(new UnifiedThread((uint)section.OwningThread));
+            Reason = UnifiedBlockingReason.CriticalSection;
+            Type = UnifiedBlockingType.CriticalSectionObject;
+            Handle = handle;
+        }
+
+        public UnifiedBlockingObject(ulong handle, string objectName, string objectType)
+            : this(OriginSource.StackWalker)
+        {
+            Owners = new List<UnifiedThread>();
+            Handle = handle;
+            KernelObjectName = objectName;
+            KernelObjectTypeName = objectType;
+            Type = UnifiedBlockingType.UnmanagedHandleObject;
+            Reason = ConvertToUnified(objectType);
+        }
+
+        public UnifiedBlockingObject(ulong handle, UnifiedBlockingType type)
+            : this(OriginSource.ThreadContextRegisters)
+        {
+            Handle = handle;
+            Type = type;
+        }
+
+        private void SetWaiters(BlockingObject item)
+        {
+            if (item.Waiters?.Count > 0)
+            {
+                Owners = new List<UnifiedThread>();
+                foreach (var waiter in item.Waiters)
+                {
+                    this.Owners.Add(new UnifiedManagedThread(waiter));
+                }
+            }
+        }
+
+        private void SetOwners(BlockingObject item)
+        {
+            if (item.Owners?.Count > 0)
+            {
+                Owners = new List<UnifiedThread>();
+                foreach (var owner in item.Owners)
+                {
+                    if (owner != null)
+                    {
+                        this.Owners.Add(new UnifiedManagedThread(owner));
+                    }
+                }
+            }
+        }
+
+        public OriginSource Origin { get; private set; }
+        public UnifiedBlockingType Type { get; private set; }
+
+        internal List<UnifiedThread> Owners { get; private set; }
+
+        public bool HasOwnershipInformation { get { return Owners != null && Owners.Count > 0; } }
+
+        public UnifiedBlockingReason Reason { get; private set; } = UnifiedBlockingReason.Unknown;
+
+        internal List<UnifiedThread> Waiters { get; private set; }
+
+        public int RecursionCount { get; private set; }
+
+        public ulong ManagedObjectAddress { get; private set; }
+
+        public string KernelObjectName { get; private set; }
+
+        public string KernelObjectTypeName { get; private set; }
+        public ulong Handle { get; private set; }
+
+        public const int BLOCK_REASON_WCT_SECTION_START_INDEX = 9;
```

Comment:
previously it was used outside the class (WaitChainTraversal), now I've changed it to private const